x-pack/plugins/security_solution/public/common/components/empty_page/index.tsx

```diff
@@ -4,84 +4,122 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiButton, EuiEmptyPrompt, EuiFlexGroup, EuiFlexItem, IconType } from '@elastic/eui';
-import React, { MouseEventHandler, ReactNode } from 'react';
+import {
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  IconType,
+  EuiCard,
+} from '@elastic/eui';
+import React, { MouseEventHandler, ReactNode, useMemo } from 'react';
 import styled from 'styled-components';
 
 const EmptyPrompt = styled(EuiEmptyPrompt)`
   align-self: center; /* Corrects horizontal centering in IE11 */
+  max-width: 60em;
 `;
 
 EmptyPrompt.displayName = 'EmptyPrompt';
 
+interface EmptyPageActions {
+  icon?: IconType;
+  label: string;
+  target?: string;
+  url: string;
+  descriptionTitle?: string;
+  description?: string;
+  fill?: boolean;
+  onClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+}
+
+export type EmptyPageActionsProps = Record<string, EmptyPageActions>;
+
 interface EmptyPageProps {
-  actionPrimaryIcon?: IconType;
-  actionPrimaryLabel: string;
-  actionPrimaryTarget?: string;
-  actionPrimaryUrl: string;
-  actionPrimaryFill?: boolean;
-  actionSecondaryIcon?: IconType;
-  actionSecondaryLabel?: string;
-  actionSecondaryTarget?: string;
-  actionSecondaryUrl?: string;
-  actionSecondaryOnClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  actions: EmptyPageActionsProps;
   'data-test-subj'?: string;
   message?: ReactNode;
   title: string;
 }
 
-export const EmptyPage = React.memo<EmptyPageProps>(
-  ({
-    actionPrimaryIcon,
-    actionPrimaryLabel,
-    actionPrimaryTarget,
-    actionPrimaryUrl,
-    actionPrimaryFill = true,
-    actionSecondaryIcon,
-    actionSecondaryLabel,
-    actionSecondaryTarget,
-    actionSecondaryUrl,
-    actionSecondaryOnClick,
-    message,
-    title,
-    ...rest
-  }) => (
+const EmptyPageComponent = React.memo<EmptyPageProps>(({ actions, message, title, ...rest }) => {
+  const titles = Object.keys(actions);
+  const renderActions = useMemo(
+    () =>
+      Object.values(actions)
+        .filter((a) => a.label && a.url)
+        .map(
+          (
+            {
+              icon,
+              label,
+              target,
+              url,
+              descriptionTitle = false,
+              description = false,
+              onClick,
+              fill = true,
+            },
+            idx
+          ) =>
+            descriptionTitle != null || description != null ? (
+              <EuiFlexItem
+                grow={false}
+                style={{ maxWidth: 283 }}
+                key={`empty-page-${titles[idx]}-action`}
+              >
+                <EuiCard
+                  title={descriptionTitle}
+                  description={description}
+                  footer={
+                    /* eslint-disable-next-line @elastic/eui/href-or-on-click */
```

Comment:
does this rule exist because the behavior is undefined if both are specified?

Reply:
It doesn't allow both href and onclick are given at the same time.